app/platform/fabric/Proxy.ts

```diff
@@ -391,4 +395,207 @@ export class Proxy {
 			);
 		}
 	}
+
+	/**
+	 *
+	 *
+	 * @param {*} channel_genesis_hash
+	 * @param {*} blockNo
+	 * @returns
+	 * @memberof Proxy
+	 */
+	async fetchDataByBlockNo(network_id, channel_genesis_hash, blockNo) {
+		const blockValue = await this.dataByBlockNo(network_id, channel_genesis_hash, blockNo);
+		if (blockValue) {
+			return blockValue;
+		}
+		else
+			return `No Block found with this block no # ${blockNo}`;
+	}
+
+	/**
+	 *
+	 *
+	 * @param {*} channel_genesis_hash
+	 * @param {*} txnId
+	 * @returns
+	 * @memberof Proxy
+	 */
+	async fetchDataByTxnId(network_id, channel_genesis_hash, txnId) {
+		let syncPlatform = new SyncPlatform(this.persistence, null)
+		let sync = new SyncServices(syncPlatform, this.persistence);
+		const client = this.platform.getClient(network_id);
+		const channel_name = client.getChannelNameByHash(channel_genesis_hash);
+		const results = await this.persistence.getCrudService().getTransactionByID(network_id, channel_genesis_hash, txnId);
+		if (results != null) {
```

Comment:
[nitpick]: Conditions can be flipped which will make it more readable.

Reply:
code will change as suggested